### DISCUSSION TITLE
How to pass in 8k data for VAD

### DISCUSSION THREAD

**Discussion by RuABraun** [snakers4/silero-vad#158] [Q&A] [answered]
Hi,
I have a question about using 8k with the `silero_vad` model.

Is the right way to pass in 8k data to the model, or should I upsample the 8k data to 16k and then pass that to the model (or is the latter method wrong)?

Thank you in advance

**Comment by snakers4** [comment, marked as answer]
Hi,

> Is the right way to pass in 8k data to the model, or should I upsample the 8k data to 16k and then pass that to the model (or is the latter method wrong)?

The `jit` model natively supports 8 kHz, just check the torch module docstring / and or see how the model is invoked in the utils.

As for the ONNX model, you have to upsample, because the ONNX model now cannot consume 8 kHz natively.

**Reply by RuABraun** [reply]
So the ONNX model will work with upsampled 8k data?

**Reply by snakers4** [reply]
The ONNX model just works only with 16k data

How its performance degrades if the data is upsampled we did not investigate, but I assume that it will be more pronounced that this - https://github.com/snakers4/silero-vad/wiki/Quality-Metrics#sample-rate-comparison

Also may depend on the upsampling method

**Reply by RuABraun** [reply]
Okay thank you

**Reply by zachllee** [reply]
@RuABraun i know this was a few years ago, how did the performance do when you upsampled?
